### ISSUE TITLE
strange char in generated file

### ISSUE THREAD

**Issue by oupala** [hacksalot/HackMyResume#168]
As discussed in this issue: mudassir0909/jsonresume-theme-elegant#73 `hackmyresume` is generating some strange char when using the `<abbr>` feature of [jsonresume-theme-elegant](mudassir0909/jsonresume-theme-elegant).

jsonresume-theme-elegant is using `markdown-it` as a markdown renderer and everything goes well when using the [test procedure](https://github.com/mudassir0909/jsonresume-theme-elegant#contributing).

But when switching to `hackmyresume`, there is some strange annoying `@@@@~` and `~@@@@` strings.

I really don't know where it does come from, except that this does not happens without `hackmyresume`. That's why I suppose it may come from `hackmyresume`.

Any tought is welcome!

**Comment by hacksalot** [comment]
I am able to reproduce this. Will be fixed for 1.9.0 or 2.0.0.

The strange characters are legacy markers that allow HackMyResume to perform arbitrary Markdown-to-HTML|MSWord|PDF|etc. on resumes generated from JSON Resume themes. There's a cleaner solution in the pipeline for 1.9.0/2.0.0, but in order to document this...

Problem: HackMyResume promises Markdown-based resume authoring under two systems: FRESH and JSON Resume. It's optimized for FRESH, but supports JSON Resume generation because we <3 JSON Resume and many users need a local-only alternative to the JSON Resume CLI tools, which were originally architected with a remote, server-side component in place.

Unfortunately, the current version of JSON Resume doesn't support Markdown natively, nor are most JSON Resume themes Markdown aware, nor for that matter do most Markdown-aware themes and tools support Markdown translation to all the formats HackMyResume supports and employers require: not just HTML but MS Word, PDF, LaTeX, and more.

So, in order to provide Markdown support even for Markdown-agnostic formats and themes under JSON Resume, when you feed it a resume with a JSON Resume theme, HackMyResume sets up guard characters around fields that need to be Markdown-transformed. It then runs the resume generation as usual (without Markdown transformation), passing the guard characters and their content to the generated document.

At this point, the generation is no different than what you'd get through JSON Resume's `resume-cli`, except that it contains some guard characters around Markdown fields. The final step is to replace each guarded field with its Markdown-translated equivalent for that format (HTML, Word, LaTeX, etc.), minus the guard characters.

What do the guard characters look like? Just as you see. `@@@@`. Again, this is legacy/interim code affecting 1.8.0 and previous. For 1.9.0 we're perform source-side translation, per-format, so this isn't necessary.

**Comment by oupala** [comment]
I tested a new resume generation with version 1.9.0-beta and it seems to work well.

No more `@@@@` in generated files!

I'm hardly waiting for the final release!

```diff
-                                                    <p class="text-muted"><small><span class="space-right">@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~Jan~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@, 2006 - @@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~Mar~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@~@@@@, 2010</span><span><i class="icon-clock mr-5"></i>4 years 3 months</span></small>
+                                                    <p class="text-muted"><small><span class="space-right">Jan, 2006 - Mar, 2010</span><span><i class="icon-clock mr-5"></i>4 years 3 months</span></small>

```
